plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3165,6 +3165,38 @@ public void testOptimizeTableAfterDeleteWithFormatVersion2()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testOptimizeCleansUpDeleteFiles()
+            throws IOException
+    {
+        String tableName = "test_optimize_" + randomTableSuffix();
+        Session sessionWithShortRetentionUnlocked = prepareCleanUpSession();
+        assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['regionkey']) AS SELECT * FROM nation", 25);
+
+        List<String> allDataFilesInitially = getAllDataFilesFromTableDirectory(tableName);
+        assertThat(allDataFilesInitially).hasSize(5);
+
+        assertUpdate("DELETE FROM " + tableName + " WHERE nationkey = 7", 1);
+
+        // Verify that delete files exists
+        List<String> allDataFilesAfterDelete = getAllDataFilesFromTableDirectory(tableName);
+        assertThat(allDataFilesAfterDelete).hasSize(6);
+
+        computeActual("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
+        computeActual(sessionWithShortRetentionUnlocked, "ALTER TABLE " + tableName + " EXECUTE EXPIRE_SNAPSHOTS (retention_threshold => '0s')");
+        computeActual(sessionWithShortRetentionUnlocked, "ALTER TABLE " + tableName + " EXECUTE REMOVE_ORPHAN_FILES (retention_threshold => '0s')");
```

Comment:
Here it's actually the `expire_snapshots` and `remove_orphan_files` calls that are removing the delete file from the table.

You can see the number of delete files in the current snapshot like this: https://github.com/trinodb/trino/pull/12351/files#diff-f5cd96d1ec33fae497df9613afcefff034f01c99adf961e927ccd76e637c7b81R3158

It should be 0 after the `optimize`

Reply:
Yes I agree it should show 0 and I know it doesn't. Funny thing it doesn't show 0 even after what I did here so OPTIMIZE, EXPIRE_SNAPSHOTS and REMOVE_ORPHAN_FILES which is at least weird. 
And after optmize I checked manually every metadata, manifest list and manifest files and I didn't find reference to the delete file from the most current snapshot.  So I don't really know where this value comes from